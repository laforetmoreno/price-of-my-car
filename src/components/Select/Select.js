import React from 'react';
import { func, string, arrayOf, bool } from 'prop-types';

import style from './Select.module.scss';

const Select = ({ onChange, placeholder, disabled, options = [], className }) => {
  const handleChange = e => {
    if (onChange) onChange(e.target.value);
  };

  return (
    <select
      disabled={disabled}
      className={`${style.wrapper} ${className} ${disabled && style.disabled}`}
      onChange={handleChange}
    >
      <>
        <option disabled selected hidden>
          {placeholder}
        </option>
        {options.map(option => (
          <option
            key={option.value ? option.value : option}
            value={option.value ? option.value : option}
          >
            {option.label ? option.label : option}
          </option>
        ))}
      </>
    </select>
  );
};

Select.propTypes = {
  onChange: func,
  options: arrayOf(string),
  placeholder: string,
  disabled: bool,
  className: string,
};

export default Select;
