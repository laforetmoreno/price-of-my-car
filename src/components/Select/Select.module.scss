@import '../../styles/variables';

.wrapper {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  color: $secondary-color;
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5;
  list-style: none;
  position: relative;
  display: inline-block;
  outline: 0;
  padding: 5px;
  border-radius: 4px;
  background: $muted-color;
  border: 1px solid transparent;
  min-width: 100px;

  &:hover {
    border: 1px solid $primary-color;
  }
}

.disabled {
  opacity: 0.5;
}
