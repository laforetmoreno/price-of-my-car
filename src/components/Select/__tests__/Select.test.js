import React from 'react';
import renderer from 'react-test-renderer';
import { render, waitForElement, fireEvent } from '@testing-library/react';
import Select from '../Select';

describe('<Select />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Select placeholder="" disabled={false} options={[{ label: 'test', value: 'test' }]} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('get by id and change', async () => {
    const { getByTestId } = render(
      <Select placeholder="" disabled={false} options={[{ label: 'test', value: 'test' }]} />,
    );
    const fieldNode = await waitForElement(() => getByTestId('select'));
    fireEvent.change(fieldNode);
  });
});
