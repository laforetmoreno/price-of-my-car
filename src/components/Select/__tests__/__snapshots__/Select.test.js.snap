// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly 1`] = `
<select
  className="wrapper undefined false"
  data-testid="select"
  defaultValue="default"
  disabled={false}
  onChange={[Function]}
>
  <option
    value="default"
  >
    
  </option>
  <option
    value="test"
  >
    test
  </option>
</select>
`;
